index.d.ts

```diff
@@ -0,0 +1,63 @@
+declare module "moleculer-apollo-server" {
+	import { ServiceSchema } from "moleculer";
+	import { Config } from "apollo-server-core";
+	export {
+		GraphQLUpload,
+		GraphQLExtension,
+		gql,
+		ApolloError,
+		toApolloError,
+		SyntaxError,
+		ValidationError,
+		AuthenticationError,
+		ForbiddenError,
+		UserInputError,
+		defaultPlaygroundOptions,
+	} from "apollo-server-core";
+
+	export * from "graphql-tools";
+
+	export interface ApolloServerOptions {
+		path: string;
+		disableHealthCheck: boolean;
+		onHealthCheck: () => {};
+	}
+
+	export class ApolloServer {
+		createGraphQLServerOptions(req: any, res: any): Promise<any>;
+		createHandler(options: ApolloServerOptions): void;
+		supportsUploads(): boolean;
+		supportsSubscriptions(): boolean;
+	}
+
+	export interface GraphQLNodeResolver {
+		action: string;
+		rootParams?: {
+			[key: string]: string;
+		};
+		dataloader?: boolean;
+	}
+
+	export interface GraphQLTypeResolver {
+		[key: string]: {
+			[key: string]: GraphQLNodeResolver;
+		};
+	}
+
+	export interface ApolloServiceOptions {
+		typeDefs?: string;
+		resolvers?: GraphQLTypeResolver;
+		routeOptions: {
+			path: string;
+			cors: boolean | Object;
+			mappingPolicy: string;
+			aliases?: any;
+			bodyParsers?: any;
+		};
+		serverOptions: Config;
+	}
+
+	export function ApolloService(options: ApolloServiceOptions): ServiceSchema;
+
+	export function moleculerGql(typeString: string, ...placeholders: any[]): string;
```

Comment:
I think this should be:
`export function moleculerGql<T>(typeString: TemplateStringsArray, ...placeholders: T[]): string;`

In particular, the `TemplateStringsArray` type is a requirement or else any use of `moleculerGql` will throw up type errors.  The Generic should be inferred under most circumstances.

[Source](https://www.bookstack.cn/read/typescriptlang-3.7/spilt.63.055dccf6c3f8aae3.md)

Reply:
I think a better type that the one I provided might be:
`export function moleculerGql<T>(typeString: TemplateStringsArray | string, ...placeholders: T[]): string;`

The apollo-server-core `gql` function is defined as accepting both `TemplateStringsArray` and `string`.